oh_staff_ui/views_utils.py

```diff
@@ -117,30 +117,81 @@ def get_result_items(queryset_list: list) -> list[ProjectItem]:
     return output_projectitems_deduped
 
 
-def get_search_results(search_type: str, query: str) -> list[ProjectItem]:
+def get_search_results(
+    search_type: str, query: str, item_type_filter: str, media_file_type_filter: str
+) -> list[ProjectItem]:
     # Return a list of items matching the search.  Different searches have
     # different return types and sort orders; this unifies them for consistent
     # use in the view and template.
-    if search_type == "title":
+
+    # first, check if this is a wildcard search - if so, return all items
+    # no need to check search_type
+    if query == "*":
+        qs_results = ProjectItem.objects.all().order_by("title")
+        filtered_results = filter_non_keyword_results_queryset(
+            qs_results, item_type_filter, media_file_type_filter
+        )
+    elif search_type == "title":
         full_query = construct_keyword_query("title", query)
         qs_results = ProjectItem.objects.filter(full_query).order_by("title")
+        filtered_results = filter_non_keyword_results_queryset(
+            qs_results, item_type_filter, media_file_type_filter
+        )
     elif search_type == "ark":
         qs_results = ProjectItem.objects.filter(ark__icontains=query).order_by("ark")
+        filtered_results = filter_non_keyword_results_queryset(
+            qs_results, item_type_filter, media_file_type_filter
+        )
     elif search_type == "status":
         qs_results = ProjectItem.objects.filter(status__status=query).order_by("title")
+        filtered_results = filter_non_keyword_results_queryset(
+            qs_results, item_type_filter, media_file_type_filter
+        )
     elif search_type == "keyword":
         search_data = get_keyword_results(query)
         # This is a sorted list of ProjectItems already, so does not need conversion to list below.
         qs_results = get_result_items(search_data)
+        filtered_results = filter_keyword_results_list(
+            qs_results, item_type_filter, media_file_type_filter
+        )
+
     # Convert query results from QuerySet to list of items, if needed.
     # qs_results is already sorted as desired based on search_type.
-    if isinstance(qs_results, QuerySet):
-        results = [item for item in qs_results]
+    if isinstance(filtered_results, QuerySet):
+        results = [item for item in filtered_results]
     else:
-        results = qs_results
+        results = filtered_results
     return results
 
 
+def filter_keyword_results_list(
```

Comment:
`filter_keyword_results_list()` and `filter_non_keyword_results_queryset()` are very similar, just acting on either a list or a queryset.  Can these be combined in a reasonable way, maybe by converting queryset to list, since that's what happens at the end of `get_search_results()` anyhow?